s25rttr fails to compile with boost-1.71.0

there are actually three errors in a row: 

```In file included from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:42:
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/common/include/helpers/containerUtils.h: In instantiation of ‘auto helpers::findPred(T&, T_Predicate&&) [with T = boost::ptr_vector<AIPlayer>; T_Predicate = GameClient::ToggleHumanAIPlayer()::<lambda(const auto:1&)>]’:
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:1752:136:   required from here
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/common/include/helpers/containerUtils.h:99:30: error: ‘begin’ was not declared in this scope
     return std::find_if(begin(container), end(container), std::forward<T_Predicate>(predicate));
                         ~~~~~^~~~~~~~~~~
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/common/include/helpers/containerUtils.h:99:30: note: suggested alternatives:
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/string:51,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/stdexcept:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/array:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/tuple:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/functional:54,
                 from /usr/include/boost/config/no_tr1/functional.hpp:21,
                 from /usr/include/boost/smart_ptr/intrusive_ptr.hpp:24,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/GameCommand.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.h:22,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:19:
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/bits/range_access.h:105:37: note:   ‘std::begin’
   template<typename _Tp> const _Tp* begin(const valarray<_Tp>&);
                                     ^~~~~
In file included from /usr/include/boost/dll/config.hpp:60,
                 from /usr/include/boost/dll/shared_library.hpp:15,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/drivers/DriverWrapper.h:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/drivers/VideoDriverWrapper.h:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:39:
/usr/include/boost/filesystem/operations.hpp:1364:5: note:   ‘boost::filesystem::begin’
     begin(const recursive_directory_iterator& iter) BOOST_NOEXCEPT
     ^~~~~
In file included from /usr/include/boost/range/functions.hpp:18,
                 from /usr/include/boost/ptr_container/detail/reversible_ptr_container.hpp:30,
                 from /usr/include/boost/ptr_container/ptr_sequence_adapter.hpp:20,
                 from /usr/include/boost/ptr_container/ptr_deque.hpp:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/external/libutil/include/libutil/MessageQueue.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/NetworkPlayer.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.h:24,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:19:
/usr/include/boost/range/begin.hpp:114:61: note:   ‘boost::range_adl_barrier::begin’
 inline BOOST_DEDUCED_TYPENAME range_iterator<const T>::type begin( const T& r )
                                                             ^~~~~
In file included from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:42:
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/common/include/helpers/containerUtils.h:99:46: error: ‘end’ was not declared in this scope
     return std::find_if(begin(container), end(container), std::forward<T_Predicate>(predicate));
                                           ~~~^~~~~~~~~~~
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/common/include/helpers/containerUtils.h:99:46: note: suggested alternatives:
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/string:51,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/stdexcept:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/array:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/tuple:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/functional:54,
                 from /usr/include/boost/config/no_tr1/functional.hpp:21,
                 from /usr/include/boost/smart_ptr/intrusive_ptr.hpp:24,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/GameCommand.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.h:22,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:19:
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/bits/range_access.h:107:37: note:   ‘std::end’
   template<typename _Tp> const _Tp* end(const valarray<_Tp>&);
                                     ^~~
In file included from /usr/include/boost/dll/config.hpp:60,
                 from /usr/include/boost/dll/shared_library.hpp:15,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/drivers/DriverWrapper.h:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/drivers/VideoDriverWrapper.h:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:39:
/usr/include/boost/filesystem/operations.hpp:1367:32: note:   ‘boost::filesystem::end’
   recursive_directory_iterator end(const recursive_directory_iterator&) BOOST_NOEXCEPT
                                ^~~
In file included from /usr/include/boost/range/functions.hpp:19,
                 from /usr/include/boost/ptr_container/detail/reversible_ptr_container.hpp:30,
                 from /usr/include/boost/ptr_container/ptr_sequence_adapter.hpp:20,
                 from /usr/include/boost/ptr_container/ptr_deque.hpp:20,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/external/libutil/include/libutil/MessageQueue.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/NetworkPlayer.h:21,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.h:24,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:19:
/usr/include/boost/range/end.hpp:108:61: note:   ‘boost::range_adl_barrier::end’
 inline BOOST_DEDUCED_TYPENAME range_iterator<const T>::type end( const T& r )
                                                             ^~~
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp: In member function ‘void GameClient::ToggleHumanAIPlayer()’:
/var/tmp/portage/games-strategy/s25rttr-9999-r4/work/s25rttr-9999/libs/s25main/network/GameClient.cpp:1752:10: error: ‘void it’ has incomplete type
     auto it = helpers::findPred(game->aiPlayers_, [id = this->GetPlayerId()](const auto& player) { return player.GetPlayerId() == id; });
          ^~

```


full log: [boost-1.71.0.log.zip](https://github.com/Return-To-The-Roots/s25client/files/3707003/boost-1.71.0.log.zip)

--- a/libs/common/include/helpers/GetInsertIterator.hpp
+++ b/libs/common/include/helpers/GetInsertIterator.hpp
@@ -19,6 +19,7 @@
 #define GetInsertIterator_h__
 
 #include <boost/type_traits/make_void.hpp>
+#include <utility>
 
 namespace helpers {
 

--- a/libs/common/include/helpers/ReserveElements.hpp
+++ b/libs/common/include/helpers/ReserveElements.hpp
@@ -19,6 +19,7 @@
 #define ReserveElements_h__
 
 #include <boost/type_traits/make_void.hpp>
+#include <utility>
 
 namespace helpers {
 

--- a/libs/common/include/helpers/containerUtils.h
+++ b/libs/common/include/helpers/containerUtils.h
@@ -22,6 +22,7 @@
 
 #include <boost/type_traits/make_void.hpp>
 #include <algorithm>
+#include <iterator>
 
 namespace helpers {
 
@@ -41,7 +42,12 @@ namespace detail {
     template<class T, class U, typename = void>
     struct FindImpl
     {
-        static auto find(T& container, const U& value) { return std::find(std::begin(container), std::end(container), value); }
+        static auto find(T& container, const U& value)
+        {
+            using std::begin;
+            using std::end;
+            return std::find(begin(container), end(container), value);
+        }
     };
 
     template<class T, class U>
@@ -55,13 +61,13 @@ namespace detail {
 /// Works only for list and set as they don't invalidate other iterators, so erase is save to call inside a loop
 /// Works also for reverse iterators
 template<typename T>
-inline typename T::iterator erase(T& container, typename T::iterator it)
+typename T::iterator erase(T& container, typename T::iterator it)
 {
     return container.erase(it);
 }
 
 template<typename T>
-inline auto erase(T& container, typename T::reverse_iterator it)
+auto erase(T& container, typename T::reverse_iterator it)
 {
     return typename T::reverse_iterator(erase(container, (++it).base()));
 }
@@ -69,18 +75,22 @@ inline auto erase(T& container, typename T::reverse_iterator it)
 template<typename T, typename T_Element>
 void remove(T& container, T_Element&& element)
 {
-    container.erase(std::remove(container.begin(), container.end(), std::forward<T_Element>(element)), container.end());
+    using std::begin;
+    using std::end;
+    container.erase(std::remove(begin(container), end(container), std::forward<T_Element>(element)), end(container));
 }
 
 template<typename T, typename T_Predicate>
 void remove_if(T& container, T_Predicate&& predicate)
 {
-    container.erase(std::remove_if(container.begin(), container.end(), std::forward<T_Predicate>(predicate)), container.end());
+    using std::begin;
+    using std::end;
+    container.erase(std::remove_if(begin(container), end(container), std::forward<T_Predicate>(predicate)), end(container));
 }
 
 /// Removes the first element in a container
 template<typename T>
-inline void pop_front(T& container)
+void pop_front(T& container)
 {
     RTTR_Assert(!container.empty());
     detail::PopFrontImpl<T>::pop(container);
@@ -88,56 +98,60 @@ inline void pop_front(T& container)
 
 /// Effective implementation of find. Uses the containers find function if available
 template<typename T, typename U>
-inline auto find(T& container, const U& value)
+auto find(T& container, const U& value)
 {
     return detail::FindImpl<T, U>::find(container, value);
 }
 
 template<typename T, class T_Predicate>
-inline auto findPred(T& container, T_Predicate&& predicate)
+auto find_if(T& container, T_Predicate&& predicate)
 {
+    using std::begin;
+    using std::end;
     return std::find_if(begin(container), end(container), std::forward<T_Predicate>(predicate));
 }
 
 /// Returns true if the container contains the given value
 /// Uses the find member function if applicable otherwise uses the std::find method
 template<typename T, typename U>
-inline bool contains(const T& container, const U& value)
+bool contains(const T& container, const U& value)
 {
-    return find(container, value) != container.end();
+    using std::end;
+    return find(container, value) != end(container);
 }
 
 /// Returns true if the container contains a value matching the predicate
 template<typename T, class T_Predicate>
-inline bool containsPred(const T& container, T_Predicate&& predicate)
+bool contains_if(const T& container, T_Predicate&& predicate)
 {
-    return findPred(container, std::forward<T_Predicate>(predicate)) != end(container);
+    using std::end;
+    return find_if(container, std::forward<T_Predicate>(predicate)) != end(container);
 }
 
 /// Remove duplicate values from the given container without changing the order
 template<class T>
-inline void makeUnique(T& container)
+void makeUnique(T& container)
 {
     // Containers with less than 2 elements are always unique
     if(container.size() < 2u)
         return;
-    auto itInsert = container.begin();
+    auto itInsert = begin(container);
     // We always begin inserting at 2nd pos so skip first
     ++itInsert;
     // And now start inserting all values starting from the 2nd
     for(auto it = itInsert; it != container.end(); ++it)
     {
         // If current element is not found in [begin, insertPos) then add it at insertPos and inc insertPos
-        if(std::find(container.begin(), itInsert, *it) == itInsert)
-            *(itInsert++) = *it;
+        if(std::find(begin(container), itInsert, *it) == itInsert)
+            *(itInsert++) = std::move(*it);
     }
-    container.erase(itInsert, container.end());
+    container.erase(itInsert, end(container));
 }
 
 /// Returns the index of the given element in the container or -1 when not found
 /// Note: Only works for containers with less than 2^31 - 1 elements.
 template<class T_Container, class T_Element>
-inline int indexOf(const T_Container& container, const T_Element element)
+int indexOf(const T_Container& container, const T_Element& element)
 {
     int index = 0;
     for(const auto& curEl : container)

--- a/libs/common/include/helpers/mathFuncs.h
+++ b/libs/common/include/helpers/mathFuncs.h
@@ -24,12 +24,12 @@ namespace helpers {
 
 /// Returns the greatest common divisor of a and b
 /// That is the greatest number x with a % x == b % x == 0
-int gcd(int a, int b);
+int gcd(int a, int b) noexcept;
 /// Returns the result of "dividend / divisor" rounded to the nearest integer value
-unsigned roundedDiv(unsigned dividend, unsigned divisor);
+unsigned roundedDiv(unsigned dividend, unsigned divisor) noexcept;
 /// Clamp the value into [min, max]
 template<typename T>
-T clamp(T val, T min, T max)
+constexpr T clamp(T val, T min, T max) noexcept
 {
     if(val <= min)
         return min;
@@ -39,7 +39,7 @@ T clamp(T val, T min, T max)
         return val;
 }
 template<typename T, typename U>
-U clamp(T val, U min, U max)
+constexpr U clamp(T val, U min, U max) noexcept
 {
     using Common = std::common_type_t<T, U>;
     if(std::is_signed<T>::value && !std::is_signed<U>::value)

--- a/libs/common/include/helpers/strUtils.h
+++ b/libs/common/include/helpers/strUtils.h
@@ -27,7 +27,7 @@ namespace helpers {
 
 /// Convert to number and return true on success
 template<typename T>
-inline bool tryFromString(const std::string& value, T& outValue)
+bool tryFromString(const std::string& value, T& outValue)
 {
     try
     {
@@ -41,7 +41,7 @@ inline bool tryFromString(const std::string& value, T& outValue)
 
 /// Convert to number or return defaultValue
 template<typename T>
-inline T fromString(const std::string& value, T defaultValue)
+T fromString(const std::string& value, T defaultValue)
 {
     T result;
     if(!tryFromString(value, result))

--- a/libs/common/src/helpers/mathFuncs.cpp
+++ b/libs/common/src/helpers/mathFuncs.cpp
@@ -21,7 +21,7 @@
 #include <utility>
 
 namespace helpers {
-int gcd(int a, int b)
+int gcd(int a, int b) noexcept
 {
     a = std::abs(a);
     b = std::abs(b);
@@ -38,7 +38,7 @@ int gcd(int a, int b)
     return a;
 }
 
-unsigned roundedDiv(unsigned dividend, unsigned divisor)
+unsigned roundedDiv(unsigned dividend, unsigned divisor) noexcept
 {
     RTTR_Assert(divisor > 0);
     RTTR_Assert(dividend + (divisor / 2) >= dividend); // Overflow check

--- a/libs/s25main/GamePlayer.cpp
+++ b/libs/s25main/GamePlayer.cpp
@@ -755,7 +755,7 @@ void GamePlayer::JobNotWanted(noRoadNode* workplace, bool all)
 
 void GamePlayer::OneJobNotWanted(const Job job, noRoadNode* workplace)
 {
-    const auto it = helpers::findPred(jobs_wanted, [workplace, job](const auto& it) { return it.workplace == workplace && it.job == job; });
+    const auto it = helpers::find_if(jobs_wanted, [workplace, job](const auto& it) { return it.workplace == workplace && it.job == job; });
     if(it != jobs_wanted.end())
         jobs_wanted.erase(it);
 }

--- a/libs/s25main/NWFInfo.cpp
+++ b/libs/s25main/NWFInfo.cpp
@@ -40,7 +40,7 @@ void NWFInfo::init(unsigned nextNWF, unsigned cmdDelay)
 
 void NWFInfo::addPlayer(unsigned playerId)
 {
-    if(!helpers::containsPred(playerInfos_, [playerId](const auto& info) { return info.id == playerId; }))
+    if(!helpers::contains_if(playerInfos_, [playerId](const auto& info) { return info.id == playerId; }))
         playerInfos_.push_back(NWFPlayerInfo(playerId));
 }
 

--- a/libs/s25main/WindowManager.cpp
+++ b/libs/s25main/WindowManager.cpp
@@ -181,7 +181,7 @@ IngameWindow* WindowManager::DoShow(std::unique_ptr<IngameWindow> window, bool m
     if(!window->IsModal())
     {
         // Check for already open windows with same ID (ignoring to-be-closed windows)
-        auto itOther = helpers::findPred(windows, [wndId = window->GetID()](const auto& curWnd) {
+        auto itOther = helpers::find_if(windows, [wndId = window->GetID()](const auto& curWnd) {
             return !curWnd->ShouldBeClosed() && !curWnd->IsModal() && curWnd->GetID() == wndId;
         });
         if(itOther != windows.end())
@@ -200,7 +200,7 @@ IngameWindow* WindowManager::DoShow(std::unique_ptr<IngameWindow> window, bool m
     }
 
     // All windows are inserted before the first modal window (shown behind)
-    auto itModal = helpers::findPred(windows, [](const auto& curWnd) { return curWnd->IsModal(); });
+    auto itModal = helpers::find_if(windows, [](const auto& curWnd) { return curWnd->IsModal(); });
     // Note that if there is no other modal window it will be put at the back which is what we want
     auto* result = windows.emplace(itModal, std::move(window))->get();
 
@@ -755,11 +755,11 @@ void WindowManager::CloseMarkedIngameWnds()
 template<class T_Windows>
 void SetActiveWindowImpl(const Window& wnd, Desktop& desktop, T_Windows& windows)
 {
-    auto itWnd = helpers::findPred(windows, [&wnd](const auto& it) { return it.get() == &wnd; });
+    auto itWnd = helpers::find_if(windows, [&wnd](const auto& it) { return it.get() == &wnd; });
     if(itWnd != windows.end())
     {
         // If we have a modal window, don't make this active unless it is the top most one
-        if(&wnd != windows.back().get() && helpers::containsPred(windows, [](const auto& it) { return it->IsModal(); }))
+        if(&wnd != windows.back().get() && helpers::contains_if(windows, [](const auto& it) { return it->IsModal(); }))
         {
             return;
         }

--- a/libs/s25main/buildings/nobMilitary.cpp
+++ b/libs/s25main/buildings/nobMilitary.cpp
@@ -866,7 +866,7 @@ unsigned nobMilitary::GetSoldiersStrength() const
 bool nobMilitary::HasMaxRankSoldier() const
 {
     const unsigned maxRank = gwg->GetGGS().GetMaxMilitaryRank();
-    return helpers::containsPred(helpers::reverse(troops), [maxRank](const auto* it) { return it->GetRank() >= maxRank; });
+    return helpers::contains_if(helpers::reverse(troops), [maxRank](const auto* it) { return it->GetRank() >= maxRank; });
 }
 
 nofDefender* nobMilitary::ProvideDefender(nofAttacker* const attacker)

--- a/libs/s25main/drivers/DriverWrapper.cpp
+++ b/libs/s25main/drivers/DriverWrapper.cpp
@@ -69,7 +69,7 @@ bool DriverWrapper::Load(const DriverType dt, std::string& preference)
         return false;
 
     /// Suche, ob der Treiber dabei ist, den wir wünschen
-    const auto it = helpers::findPred(drivers, [preference](const auto& it) { return it.GetName() == preference; });
+    const auto it = helpers::find_if(drivers, [preference](const auto& it) { return it.GetName() == preference; });
     if(it != drivers.end())
     {
         // Dann den gleich nehmen

--- a/libs/s25main/ingameWindows/iwBaseWarehouse.cpp
+++ b/libs/s25main/ingameWindows/iwBaseWarehouse.cpp
@@ -218,7 +218,7 @@ void iwBaseWarehouse::Msg_ButtonClick(const unsigned ctrl_id)
             const std::list<nobBaseWarehouse*>& storehouses =
               gwv.GetWorld().GetPlayer(wh->GetPlayer()).GetBuildingRegister().GetStorehouses();
             // go through list once we get to current building -> open window for the next one and go to next location
-            auto it = helpers::findPred(storehouses, [whPos = wh->GetPos()](const auto* it) { return it->GetPos() == whPos; });
+            auto it = helpers::find_if(storehouses, [whPos = wh->GetPos()](const auto* it) { return it->GetPos() == whPos; });
             if(it != storehouses.end()) // got to current building in the list?
             {
                 // close old window, open new window (todo: only open if it isnt already open), move to location of next building

--- a/libs/s25main/ingameWindows/iwBuilding.cpp
+++ b/libs/s25main/ingameWindows/iwBuilding.cpp
@@ -225,7 +225,7 @@ void iwBuilding::Msg_ButtonClick(const unsigned ctrl_id)
             const std::list<nobUsual*>& buildings =
               gwv.GetWorld().GetPlayer(building->GetPlayer()).GetBuildingRegister().GetBuildings(building->GetBuildingType());
             // go through list once we get to current building -> open window for the next one and go to next location
-            auto it = helpers::findPred(buildings, [bldPos = building->GetPos()](const auto* it) { return it->GetPos() == bldPos; });
+            auto it = helpers::find_if(buildings, [bldPos = building->GetPos()](const auto* it) { return it->GetPos() == bldPos; });
             if(it != buildings.end()) // got to current building in the list?
             {
                 // close old window, open new window (todo: only open if it isnt already open), move to location of next building

--- a/libs/s25main/ingameWindows/iwMilitaryBuilding.cpp
+++ b/libs/s25main/ingameWindows/iwMilitaryBuilding.cpp
@@ -197,8 +197,7 @@ void iwMilitaryBuilding::Msg_ButtonClick(const unsigned ctrl_id)
             const std::list<nobMilitary*>& militaryBuildings =
               gwv.GetWorld().GetPlayer(building->GetPlayer()).GetBuildingRegister().GetMilitaryBuildings();
             // go through list once we get to current building -> open window for the next one and go to next location
-            auto it =
-              helpers::findPred(militaryBuildings, [bldPos = building->GetPos()](const auto* it) { return it->GetPos() == bldPos; });
+            auto it = helpers::find_if(militaryBuildings, [bldPos = building->GetPos()](const auto* it) { return it->GetPos() == bldPos; });
             if(it != militaryBuildings.end()) // got to current building in the list?
             {
                 // close old window, open new window (todo: only open if it isnt already open), move to location of next building

--- a/libs/s25main/ingameWindows/iwTextfile.cpp
+++ b/libs/s25main/ingameWindows/iwTextfile.cpp
@@ -49,7 +49,7 @@ iwTextfile::iwTextfile(const std::string& filename, const std::string& title)
         paths.push_back(basePath / folderName / filename);
     paths.push_back(basePath / filename);
 
-    const auto existingFilePath = helpers::findPred(paths, [](const auto& path) { return bfs::exists(path); });
+    const auto existingFilePath = helpers::find_if(paths, [](const auto& path) { return bfs::exists(path); });
     bnw::ifstream file;
     if(existingFilePath != paths.end())
         file.open(*existingFilePath);

--- a/libs/s25main/network/GameClient.cpp
+++ b/libs/s25main/network/GameClient.cpp
@@ -1749,7 +1749,7 @@ unsigned GameClient::GetTournamentModeDuration() const
 void GameClient::ToggleHumanAIPlayer()
 {
     RTTR_Assert(!IsReplayModeOn());
-    auto it = helpers::findPred(game->aiPlayers_, [id = this->GetPlayerId()](const auto& player) { return player.GetPlayerId() == id; });
+    auto it = helpers::find_if(game->aiPlayers_, [id = this->GetPlayerId()](const auto& player) { return player.GetPlayerId() == id; });
     if(it != game->aiPlayers_.end())
         game->aiPlayers_.erase(it);
     else

--- a/libs/s25main/pathfinding/FreePathFinder.cpp
+++ b/libs/s25main/pathfinding/FreePathFinder.cpp
@@ -222,7 +222,7 @@ bool FreePathFinder::FindPathAlternatingConditions(const MapPoint start, const M
                         back_id = alternate ? nodes[back_id].prevEven : nodes[back_id].prev;
                         alternate = !alternate;
                     }
-                    bool tooClose = helpers::containsPred(
+                    bool tooClose = helpers::contains_if(
                       evenLocationsOnRoute, [this, neighbourPos](const MapPoint& it) { return gwb_.CalcDistance(neighbourPos, it) < 2; });
                     if(tooClose)
                         continue;

